src/api/transaction/sign.js

```diff
@@ -17,20 +18,16 @@ const validate = utils.common.validate;
  */
 const HASH_TX_ID = 0x54584E00; // 'TXN'
 
-function serialize(txJSON) {
-  return core.SerializedObject.from_json(txJSON);
-}
-
 function hashSerialization(serialized, prefix) {
-  return serialized.hash(prefix || HASH_TX_ID).to_hex();
-}
-
-function signingData(txJSON) {
-  return core.Transaction.from_json(txJSON).signingData().buffer;
+  const hexPrefix = prefix.toString(16).toUpperCase();
+  const toHash = hexPrefix + serialized;
+  return hashjs.sha512().update(new Buffer(toHash, 'hex')).digest('hex')
+    .toUpperCase().slice(0, 64);
```

Comment:
Yeah, that's a ripple pattern: "For a hash function, rippled uses SHA-512 and then truncates the result to the first 256 bytes. This algorithm, informally called SHA-512Half, provides an output that has comparable security to SHA-256, but runs faster on 64-bit processors." https://ripple.com/build/ledger-format/

I'll factor out a the sha512hash function and add a comment based on this quote from the docs.